runelite-client/src/main/java/net/runelite/client/plugins/puzzlesolver/solver/pathfinding/IDAStarMM.java

```diff
@@ -0,0 +1,838 @@
+					if (locVal.getX() - 1 == locTarget.getX())
+					{
+						//Swap blank to right and continue
+						Point loc1 = new Point(locBlank.getX() + 1, locBlank.getY());
+						Point loc2 = new Point(loc1.getX(), loc1.getY() + 1);
+
+						swap(locBlank, loc1);
+						swap(loc1, loc2);
+
+						continue;
+					}
+
+					Point loc1 = new Point(locBlank.getX() - 1, locBlank.getY());
+					Point loc2 = new Point(loc1.getX(), loc1.getY() + 1);
+
+					swap(locBlank, loc1);
+					swap(loc1, loc2);
+					swap(loc2, locVal);
+				}
+			}
+			else if (locVal.getY() == locBlank.getY())
+			{
+				int diff = locBlank.getX() - locVal.getX();
+				if (diff == 1)
+				{
+					//Right
+
+					//Last piece
+					if (locTarget.getY() == DIMENSION - 1)
+					{
+						Point loc1 = new Point(locVal.getX() - 1, locVal.getY() - 1);
+						Point loc2 = new Point(locVal.getX() - 1, locVal.getY());
+						Point loc3 = new Point(locVal.getX(), locVal.getY() - 1);
+						Point loc4 = new Point(locVal.getX() + 1, locVal.getY() - 1);
+
+						swap(locBlank, locVal);
+
+						swap(locVal, loc3);
+						swap(loc3, loc1);
+						swap(loc1, loc2);
+						swap(loc2, locVal);
+						swap(locVal, loc3);
+						swap(loc3, loc1);
+						swap(loc1, loc2);
+						swap(loc2, locVal);
+						swap(locVal, locBlank);
+						swap(locBlank, loc4);
+						swap(loc4, loc3);
+						swap(loc3, loc1);
+						swap(loc1, loc2);
+						swap(loc2, locVal);
+
+						return;
+					}
+
+					Point loc1 = new Point(locVal.getX() + 1, locVal.getY() + 1);
+					Point loc2 = new Point(locVal.getX(), locVal.getY() + 1);
+					Point loc3 = new Point(locVal.getX() - 1, locVal.getY() + 1);
+					Point loc4 = new Point(locVal.getX() - 1, locVal.getY());
+
+					swap(locBlank, loc1);
+					swap(loc1, loc2);
+					swap(loc2, loc3);
+					swap(loc3, loc4);
+					swap(loc4, locVal);
+				}
+				else if (diff == -1)
+				{
+					//Left
+					swap(locBlank, locVal);
+				}
+			}
+		}
+	}
+
+	private void swap(Point p1, Point p2)
+	{
+		int[] pieces = currentState.getPuzzle().clone();
+
+		int val1 = getPiece(p1);
+		int val2 = getPiece(p2);
+
+		if (!checkSwap(p1, p2))
+		{
+			throw new IllegalStateException(String.format("Invalid swap: %1$s, %2$s", p1.toString(), p2.toString()));
+		}
+
+		pieces[p1.getY() * DIMENSION + p1.getX()] = val2;
+		pieces[p2.getY() * DIMENSION + p2.getX()] = val1;
+
+		currentState = new PuzzleState(pieces);
+		stateList.add(new PuzzleState(currentState));
+	}
+
+	private boolean checkSwap(Point p1, Point p2)
+	{
+		int absX = Math.abs(p1.getX() - p2.getX());
+		int absY = Math.abs(p1.getY() - p2.getY());
+
+		if (getPiece(p1) != BLANK_TILE_VALUE && getPiece(p2) != BLANK_TILE_VALUE)
+		{
+			return false;
+		}
+
+		if (p1.getX() == p2.getX() && absY == 1)
+		{
+			return true;
+		}
+
+		if (p1.getY() == p2.getY() && absX == 1)
+		{
+			return true;
+		}
+
+		return false;
+	}
+
+	private int getPiece(Point p)
+	{
+		return currentState.getPiece(p.getX(), p.getY());
```

Comment:
There's liberal usage of Point all over. I think you can just pass in x and y as ints instead of packing it into Point everywhere.